tensorflow/python/ops/linalg/linalg_impl.py

```diff
@@ -276,16 +276,16 @@ def matrix_exponential(input, name=None):  # pylint: disable=redefined-builtin
         math_ops.reduce_sum(
             math_ops.abs(matrix),
             axis=array_ops.size(array_ops.shape(matrix)) - 2),
-        axis=-1)
+        axis=-1)[..., array_ops.newaxis, array_ops.newaxis]
```

Comment:
I think expand_dims will read a little better than newaxis here, but up to you.

Reply:
Unless I'm missing a nicer way to do it, I think newaxis is a little cleaner (because multiple expan_dims would be necessary). With expand_dims I think it would be:
```
l1_norm = array_ops.expand_dims(array_ops.expand_dims(math_ops.reduce_max(
        math_ops.reduce_sum(
            math_ops.abs(matrix),
            axis=array_ops.size(array_ops.shape(matrix)) - 2),
        axis=-1), -1), -1)
```
I suppose there'd also be:
```
l1_norm_flat = ...
l1_norm = array_ops.expand_dims(array_ops.expand_dims(l1_norm_flat, -1), -1)
```

To me both of those are a bit less intuitive than using newaxis, although that might just be because I'm more familiar with newaxis.